### ISSUE TITLE
Abstract subclasses doesn't work: issubclass() arg 1 must be a class

### ISSUE THREAD

**Issue by alexgleason** [torchbox/wagtail#1603]
Here's an example:

```python
class SitePage(Page):
    pass
    class Meta:
        abstract = True

class HomePage(SitePage):
    pass
```

This causes the following error:

```
Environment:


Request Method: GET
Request URL: http://localhost:8000/admin/

Django Version: 1.8.2
Python Version: 2.7.6
Installed Applications:
('django.contrib.admin',
 'django.contrib.auth',
 'django.contrib.contenttypes',
 'django.contrib.sessions',
 'django.contrib.messages',
 'django.contrib.staticfiles',
 'taggit',
 'compressor',
 'modelcluster',
 'wagtail.wagtailcore',
 'wagtail.wagtailadmin',
 'wagtail.wagtailsearch',
 'wagtail.wagtailimages',
 'wagtail.wagtaildocs',
 'wagtail.wagtailsnippets',
 'wagtail.wagtailusers',
 'wagtail.wagtailsites',
 'wagtail.wagtailembeds',
 'wagtail.wagtailredirects',
 'wagtail.wagtailforms',
 'pennebaker_wagtail',
 'search',
 'journal')
Installed Middleware:
('django.contrib.sessions.middleware.SessionMiddleware',
 'django.middleware.common.CommonMiddleware',
 'django.middleware.csrf.CsrfViewMiddleware',
 'django.contrib.auth.middleware.AuthenticationMiddleware',
 'django.contrib.auth.middleware.SessionAuthenticationMiddleware',
 'django.contrib.messages.middleware.MessageMiddleware',
 'django.middleware.clickjacking.XFrameOptionsMiddleware',
 'django.middleware.security.SecurityMiddleware',
 'wagtail.wagtailcore.middleware.SiteMiddleware',
 'wagtail.wagtailredirects.middleware.RedirectMiddleware')


Template error:
In template /usr/local/lib/python2.7/dist-packages/wagtail/wagtailadmin/templates/wagtailadmin/base.html, error at line 22
   issubclass() arg 1 must be a class

   12 :             </a>



   13 : 



   14 :             <form class="nav-search" action="{% url 'wagtailadmin_pages_search' %}" method="get">



   15 :                 <div>



   16 :                     <label for="menu-search-q">{% trans "Search" %}</label>



   17 :                     <input type="text" id="menu-search-q" name="q" placeholder="{% trans 'Search' %}" />



   18 :                     <button type="submit">{% trans "Search" %}</button>



   19 :                 </div>



   20 :             </form>



   21 : 



   22 :              {% main_nav %} 



   23 :         </div>



   24 : 



   25 :         <span class="explorer-close">Close explorer</span>



   26 :         <nav id="explorer" class="explorer">



   27 :         </nav>



   28 :     </div>



   29 : 



   30 :     <div class="content-wrapper">



   31 :         <div class="content">



   32 :             {# Always show messages div so it can be appended to by JS #}


Traceback:
File "/usr/local/lib/python2.7/dist-packages/django/core/handlers/base.py" in get_response
  132.                     response = wrapped_callback(request, *callback_args, **callback_kwargs)
File "/usr/local/lib/python2.7/dist-packages/django/views/decorators/cache.py" in _cache_controlled
  43.             response = viewfunc(request, *args, **kw)
File "/usr/local/lib/python2.7/dist-packages/django/contrib/auth/decorators.py" in _wrapped_view
  22.                 return view_func(request, *args, **kwargs)
File "/usr/local/lib/python2.7/dist-packages/wagtail/wagtailadmin/views/home.py" in home
  78.         'user': request.user
File "/usr/local/lib/python2.7/dist-packages/django/shortcuts.py" in render
  67.             template_name, context, request=request, using=using)
File "/usr/local/lib/python2.7/dist-packages/django/template/loader.py" in render_to_string
  99.         return template.render(context, request)
File "/usr/local/lib/python2.7/dist-packages/django/template/backends/django.py" in render
  74.         return self.template.render(context)
File "/usr/local/lib/python2.7/dist-packages/django/template/base.py" in render
  209.                     return self._render(context)
File "/usr/local/lib/python2.7/dist-packages/django/template/base.py" in _render
  201.         return self.nodelist.render(context)
File "/usr/local/lib/python2.7/dist-packages/django/template/base.py" in render
  903.                 bit = self.render_node(node, context)
File "/usr/local/lib/python2.7/dist-packages/django/template/debug.py" in render_node
  79.             return node.render(context)
File "/usr/local/lib/python2.7/dist-packages/django/template/loader_tags.py" in render
  135.         return compiled_parent._render(context)
File "/usr/local/lib/python2.7/dist-packages/django/template/base.py" in _render
  201.         return self.nodelist.render(context)
File "/usr/local/lib/python2.7/dist-packages/django/template/base.py" in render
  903.                 bit = self.render_node(node, context)
File "/usr/local/lib/python2.7/dist-packages/django/template/debug.py" in render_node
  79.             return node.render(context)
File "/usr/local/lib/python2.7/dist-packages/django/template/loader_tags.py" in render
  135.         return compiled_parent._render(context)
File "/usr/local/lib/python2.7/dist-packages/django/template/base.py" in _render
  201.         return self.nodelist.render(context)
File "/usr/local/lib/python2.7/dist-packages/django/template/base.py" in render
  903.                 bit = self.render_node(node, context)
File "/usr/local/lib/python2.7/dist-packages/django/template/debug.py" in render_node
  79.             return node.render(context)
File "/usr/local/lib/python2.7/dist-packages/django/template/loader_tags.py" in render
  135.         return compiled_parent._render(context)
File "/usr/local/lib/python2.7/dist-packages/django/template/base.py" in _render
  201.         return self.nodelist.render(context)
File "/usr/local/lib/python2.7/dist-packages/django/template/base.py" in render
  903.                 bit = self.render_node(node, context)
File "/usr/local/lib/python2.7/dist-packages/django/template/debug.py" in render_node
  79.             return node.render(context)
File "/usr/local/lib/python2.7/dist-packages/django/template/loader_tags.py" in render
  65.                 result = block.nodelist.render(context)
File "/usr/local/lib/python2.7/dist-packages/django/template/base.py" in render
  903.                 bit = self.render_node(node, context)
File "/usr/local/lib/python2.7/dist-packages/django/template/debug.py" in render_node
  79.             return node.render(context)
File "/usr/local/lib/python2.7/dist-packages/django/template/base.py" in render
  1271.                     _dict = func(*resolved_args, **resolved_kwargs)
File "/usr/local/lib/python2.7/dist-packages/wagtail/wagtailadmin/templatetags/wagtailadmin_tags.py" in main_nav
  38.         'menu_html': admin_menu.render_html(request),
File "/usr/local/lib/python2.7/dist-packages/wagtail/wagtailadmin/menu.py" in render_html
  83.         menu_items = self.menu_items_for_request(request)
File "/usr/local/lib/python2.7/dist-packages/wagtail/wagtailadmin/menu.py" in menu_items_for_request
  70.         return [item for item in self.registered_menu_items if item.is_shown(request)]
File "/usr/local/lib/python2.7/dist-packages/wagtail/wagtailforms/wagtail_hooks.py" in is_shown
  21.         return get_forms_for_user(request.user).exists()
File "/usr/local/lib/python2.7/dist-packages/wagtail/wagtailforms/models.py" in get_forms_for_user
  118.     return editable_pages.filter(content_type__in=get_form_types())
File "/usr/local/lib/python2.7/dist-packages/wagtail/wagtailforms/models.py" in get_form_types
  110.             if issubclass(ct.model_class(), AbstractForm)

Exception Type: TypeError at /admin/
Exception Value: issubclass() arg 1 must be a class

```

**Comment by alexgleason** [comment]
I added a PR to this. Please let me know if this is user error, as I'm still learning Django.

**Comment by alexgleason** [comment]
Just tried it using virtualenv with the latest version of python and did get the same error.

To be clear, I'm trying to make an abstract model containing fields that all pages on the site will have. For instance, all pages on my site have a banner at the top. Here's a more complete example of my code:

```python
class SitePage(Page):
    banner = models.ForeignKey(
        'wagtailimages.Image',
        null=True,
        blank=True,
        on_delete=models.SET_NULL,
        related_name='+'
    )
    content_panels = Page.content_panels + [
        ImageChooserPanel('banner')
    ]

class AboutPage(SitePage):
    team_members = StreamField([...])
```

I'm wanting AboutPage to have the banner, but then have its own fields.

As a workaround for now, I've simply left them all inheriting Page and copied the fields into them.

**Comment by alexgleason** [comment]
This code works:

```python
class SitePage(models.Model):
    banner = models.ForeignKey([...])
    class Meta:
        abstract = True

class AboutPage(Page, SitePage):
    team_members = StreamField([...])
```

But unfortunately I can't set content_panels of the base class because it's a Django model and not a Wagtail page, I believe.

**Comment by timheap** [comment]
Try adding the `is_abstract` field to your base model class:

```python
class SitePage(models.Model): # using Django model for base class
    banner = models.ForeignKey([...])
    is_abstract = True
    class Meta:
        abstract = True
```

Sadly, this attribute does not seem to be documented any more. Perhaps it got misplaced when the documentation was being rearranged, or perhaps it was never documented. Either way, Wagtail does not actually look at the `_meta.abstract` field. If you want an abstract page type, you have to set both `MyPage.is_abstract = True`, AND `MyPage._meta.abstract = True`.

**Comment by alexgleason** [comment]
confirmed! `is_abstract = True` solved this.

**Comment by kaedroho** [comment]
Thanks for confirming! This is definitely something we need to fix in the docs.

Is there no way we can make whatever needs ``is_abstract`` to be set to look at ``_meta.abstract`` instead?

**Comment by gasman** [comment]
`is_abstract` was added for a different purpose than `_meta.abstract` - it indicates that users are not meant to create that type of page directly (and so it shouldn't show up in the 'create page' list). In particular, the wagtailcore.Page model is marked as `is_abstract`, but isn't an abstract model in the Django sense.

**Comment by kaedroho** [comment]
Maybe we could make it look at both then:

```python
page_is_abstract = page.is_abstract or page._meta.abstract
```

**Comment by gasman** [comment]
Sounds like a good plan to me.

**Comment by gasman** [comment]
Fixed in #1631 - thanks for the initial investigation and PR, @alexgleason!

**Comment by alexgleason** [comment]
Awesome, thanks for the great work!
